lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1630,7 +1630,7 @@ std::vector<std::string> SessionState::get_final_action_restrict_rules(
     const CreditKey& charging_key) const {
   std::vector<std::string> restrict_rules;
   auto it = credit_map_.find(charging_key);
-  if (it != credit_map_.end()) {
+  if (it == credit_map_.end()) {
     return restrict_rules;
   }
   restrict_rules = it->second->final_action_info.restrict_rules;
```

Comment:
@themarwhal It will not work here. 
We will hit the copy elision optimisation by gcc preventing the vector from being copied.
This would work perfectly if we were using the returned value immediately.

Unfortunately, restrict_rules is used after receive_charging_credit which overrides the old values of final action. 
I reworked the function to pass a reference instead of returning a vector.

Reply:
:'( ok!